tests/Features/ReturnTypes/CustomEloquentCollectionTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Tests\Features\ReturnTypes;
+
+use App\Account;
+use App\AccountCollection;
+
+class CustomEloquentCollectionTest
+{
+    public function testQueryBuilderHydrateReturnsCustomBuilder(): AccountCollection
```

Comment:
You can add a `return` docblock here that will look something like `@phpstan-return AccountCollection<Account>` So that way we can assert it returns correct Collection class with the correct model.

(This applies to other tests where AccountCollection is returned)